File: src/main/java/de/dk/api/codeChallenge/notification/model/NotificationTypeEntity.java
package de.dk.api.codeChallenge.notification.model;

import de.dk.api.codeChallenge.notification.model.enums.Category;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.util.UUID;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "notification_types")
@Getter
@Setter
@NoArgsConstructor
public class NotificationTypeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "uuid")
    private UUID id;

    @Column(nullable = false, length = 64)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 16)
    private Category category;

    public NotificationTypeEntity(String name, Category category) {
        this.name = name;
        this.category = category;
    }
}

File: src/main/kotlin/de/dkb/api/codeChallenge/notification/model/NotificationTypeEntity.kt
package de.dkb.api.codeChallenge.notification.model

import de.dkb.api.codeChallenge.notification.model.enums.Category
import jakarta.persistence.*
import org.hibernate.annotations.UuidGenerator
import java.util.UUID

@Entity
@Table(
    name = "notification_types"
)
class NotificationTypeEntity() {

    @Id
    @UuidGenerator
    @Column(columnDefinition = "uuid")
    var id: UUID? = null

    @Column(nullable = false, length = 64)
    lateinit var name: String

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 16)
    lateinit var category: Category

    constructor(name: String, category: Category) : this() {
        this.name = name
        this.category = category
    }
}


